cypari2/convert.pyx

```diff
@@ -104,63 +115,70 @@ cpdef gen_to_integer(Gen x):
       real ``t_COMPLEX``, a ``t_INTMOD`` or ``t_PADIC`` (which are
       lifted).
 
-    EXAMPLES::
-
-        sage: from sage.libs.cypari2.convert import gen_to_integer
-        sage: a = gen_to_integer(pari("12345")); a; type(a)
-        12345
-        <... 'int'>
-        sage: a = gen_to_integer(pari("10^30")); a; type(a)
-        1000000000000000000000000000000L
-        <... 'long'>
-        sage: gen_to_integer(pari("19/5"))
-        3
-        sage: gen_to_integer(pari("1 + 0.0*I"))
-        1
-        sage: gen_to_integer(pari("3/2 + 0.0*I"))
-        1
-        sage: gen_to_integer(pari("Mod(-1, 11)"))
-        10
-        sage: gen_to_integer(pari("5 + O(5^10)"))
-        5
-        sage: gen_to_integer(pari("Pol(42)"))
-        42
-        sage: gen_to_integer(pari("x"))
-        Traceback (most recent call last):
-        ...
-        TypeError: unable to convert PARI object x of type t_POL to an integer
-        sage: gen_to_integer(pari("x + O(x^2)"))
-        Traceback (most recent call last):
-        ...
-        TypeError: unable to convert PARI object x + O(x^2) of type t_SER to an integer
-        sage: gen_to_integer(pari("1 + I"))
-        Traceback (most recent call last):
-        ...
-        TypeError: unable to convert PARI object 1 + I of type t_COMPLEX to an integer
-
-    TESTS::
-
-        sage: for i in range(10000):
-        ....:     x = 3**i
-        ....:     if long(pari(x)) != long(x):
-        ....:         print(x)
-        sage: gen_to_integer(pari("1.0 - 2^64"))
-        -18446744073709551615L
-        sage: gen_to_integer(pari("1 - 2^64"))
-        -18446744073709551615L
-
-    Check some corner cases::
-
-        sage: for s in [1, -1]:
-        ....:     for a in [1, 2**31, 2**32, 2**63, 2**64]:
-        ....:         for b in [-1, 0, 1]:
-        ....:             Nstr = str(s * (a + b))
-        ....:             N1 = gen_to_integer(pari(Nstr))  # Convert via PARI
-        ....:             N2 = int(Nstr)                   # Convert via Python
-        ....:             if N1 != N2:
-        ....:                 print(Nstr, N1, N2)
-        ....:             if type(N1) is not type(N2):
-        ....:                 print(N1, type(N1), N2, type(N2))
+    Examples:
+
+    >>> from cypari2.convert import gen_to_integer
+    >>> from cypari2 import Pari
+    >>> pari = Pari()
+    >>> a = gen_to_integer(pari("12345")); a; type(a)
+    12345
+    <... 'int'>
+    >>> gen_to_integer(pari("10^30")) == 10**30
+    True
+    >>> gen_to_integer(pari("19/5"))
+    3
+    >>> gen_to_integer(pari("1 + 0.0*I"))
+    1
+    >>> gen_to_integer(pari("3/2 + 0.0*I"))
+    1
+    >>> gen_to_integer(pari("Mod(-1, 11)"))
+    10
+    >>> gen_to_integer(pari("5 + O(5^10)"))
+    5
+    >>> gen_to_integer(pari("Pol(42)"))
+    42
+    >>> gen_to_integer(pari("u"))
```

Comment:
No idea. This is still failing on travis. However the option `IGNORE_EXCEPTION_DETAIL` in doctests make this failure disappear.